tests/modules/happy/test.yml

```diff
@@ -0,0 +1,27 @@
+- name: happy test_happy_vcf
+  command: nextflow run tests/modules/happy -entry test_happy_vcf -c tests/config/nextflow.config
+  tags:
+    - happy
+  files:
+    - path: output/happy/test.extended.csv
+      md5sum: ef79c7c789ef4f146ca2e50dafaf22b3
+    - path: output/happy/test.runinfo.json
+      contains: dist
```

Comment:
Can you `diff` both the output from both tries to see what part of the file is variable?

Reply:
```
$ diff happy/test.runinfo.json /tmp/tmp87deay82/happy/test.runinfo.json
1c1
< {"dist": " /  / ", "python_version": "2.7.15", "timestamp": "Wed May 04 12:21:51 2022", "python_implementation": "CPython", "uname": "Linux / dbf3e7b09ea1 / 5.14.0-1034-oem / #37-Ubuntu SMP Tue Apr 19 14:13:20 UTC 2022 / x86_64 / ", "python_prefix": "/usr/local", "environment": {"SHLVL": "2", "HOSTNAME": "dbf3e7b09ea1", "DYLD_LIBRARY_PATH": "/usr/local/lib", "PWD": "/tmp/tmpjrhup01o/ce/f45530c174dc7f002a265e66653556", "HOME": "/home/nvnieuwk", "PATH": "/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", "LD_LIBRARY_PATH": "/usr/local/lib", "_": "/usr/local/bin/hap.py"}, "name": "hap.py", "mac_ver": " / ", "final_args": {"engine": "xcmp", "verbose": false, "roc": "QUAL", "locations": null, "write_counts": true, "preprocessing_truth_confregions": true, "preprocessing_leftshift": true, "filters_only": "", "type": null, "do_roc": true, "vcf2": "test_preprocessed.vcf.gz", "engine_vcfeval": "rtg", "force_interactive": true, "vcf1": "test.rnaseq.vcf", "hb_expand": 30, "engine_vcfeval_template": null, "convert_gvcf_query": false, "strat_tsv": null, "write_vcf": true, "window": 50, "somatic_allele_conversion": false, "roc_filter": false, "version": false, "ref": "genome.fasta", "fp_bedfile": null, "filter_nonref": false, "_vcfs": ["test.rnaseq.vcf", "test_preprocessed.vcf.gz"], "max_enum": 16768, "engine_scmp_distance": 30, "convert_gvcf_to_vcf": false, "no_hc": false, "usefiltered_truth": false, "regions_bedfile": "genome.bed", "fixchr": null, "strat_fixchr": null, "preserve_info": false, "logfile": null, "pass_only": false, "scratch_prefix": null, "gender": "auto", "output_vtc": false, "preprocessing_decompose": true, "convert_gvcf_truth": false, "ci_alpha": 0.0, "bcf": false, "delete_scratch": true, "quiet": false, "write_json": true, "preprocess_window": 10000, "roc_delta": 0.5, "targets_bedfile": null, "roc_regions": ["*"], "reports_prefix": "test", "strat_regions": [], "threads": 2, "preprocessing_truth": false, "preprocessing_norm": false}, "runInfo": [{"value": "/usr/local/bin/hap.py test.rnaseq.vcf test_preprocessed.vcf.gz --reference genome.fasta --threads 2 -R genome.bed -o test", "key": "commandline"}], "version": "", "metadata": {"required": {"version": "", "id": "haplotypes", "module": "hap.py", "description": "/usr/local/bin/hap.py generated this JSON file via command line /usr/local/bin/hap.py test.rnaseq.vcf test_preprocessed.vcf.gz --reference genome.fasta --threads 2 -R genome.bed -o test"}}}
\ No newline at end of file
---
> {"dist": " /  / ", "python_version": "2.7.15", "timestamp": "Wed May 04 12:21:33 2022", "python_implementation": "CPython", "uname": "Linux / e89638307e70 / 5.14.0-1034-oem / #37-Ubuntu SMP Tue Apr 19 14:13:20 UTC 2022 / x86_64 / ", "python_prefix": "/usr/local", "environment": {"SHLVL": "2", "HOSTNAME": "e89638307e70", "DYLD_LIBRARY_PATH": "/usr/local/lib", "PWD": "/tmp/tmpw5l8fi_7/74/2463b752f033e4e27c7fe29fc35646", "HOME": "/home/nvnieuwk", "PATH": "/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", "LD_LIBRARY_PATH": "/usr/local/lib", "_": "/usr/local/bin/hap.py"}, "name": "hap.py", "mac_ver": " / ", "final_args": {"engine": "xcmp", "verbose": false, "roc": "QUAL", "locations": null, "write_counts": true, "preprocessing_truth_confregions": true, "preprocessing_leftshift": true, "filters_only": "", "type": null, "do_roc": true, "vcf2": "test_preprocessed.vcf.gz", "engine_vcfeval": "rtg", "force_interactive": true, "vcf1": "test.rnaseq.vcf", "hb_expand": 30, "engine_vcfeval_template": null, "convert_gvcf_query": false, "strat_tsv": null, "write_vcf": true, "window": 50, "somatic_allele_conversion": false, "roc_filter": false, "version": false, "ref": "genome.fasta", "fp_bedfile": null, "filter_nonref": false, "_vcfs": ["test.rnaseq.vcf", "test_preprocessed.vcf.gz"], "max_enum": 16768, "engine_scmp_distance": 30, "convert_gvcf_to_vcf": false, "no_hc": false, "usefiltered_truth": false, "regions_bedfile": "genome.bed", "fixchr": null, "strat_fixchr": null, "preserve_info": false, "logfile": null, "pass_only": false, "scratch_prefix": null, "gender": "auto", "output_vtc": false, "preprocessing_decompose": true, "convert_gvcf_truth": false, "ci_alpha": 0.0, "bcf": false, "delete_scratch": true, "quiet": false, "write_json": true, "preprocess_window": 10000, "roc_delta": 0.5, "targets_bedfile": null, "roc_regions": ["*"], "reports_prefix": "test", "strat_regions": [], "threads": 2, "preprocessing_truth": false, "preprocessing_norm": false}, "runInfo": [{"value": "/usr/local/bin/hap.py test.rnaseq.vcf test_preprocessed.vcf.gz --reference genome.fasta --threads 2 -R genome.bed -o test", "key": "commandline"}], "version": "", "metadata": {"required": {"version": "", "id": "haplotypes", "module": "hap.py", "description": "/usr/local/bin/hap.py generated this JSON file via command line /usr/local/bin/hap.py test.rnaseq.vcf test_preprocessed.vcf.gz --reference genome.fasta --threads 2 -R genome.bed -o test"}}}
\ No newline at end of file
```

I only notice some small differences like date and PWD, but these will be variable every time the test is repeated

Reply:
Then you can omit the `md5sum` or `contains` and just check for the file's existence (as you do now)